internal/services/network/virtual_network_gateway_connection_resource.go

```diff
@@ -78,7 +78,7 @@ func resourceVirtualNetworkGatewayConnection() *pluginsdk.Resource {
 
 			"shared_key": {
 				Type:      pluginsdk.TypeString,
-				Required:  true,
+				Optional:  true,
```

Comment:
This was previously `Computed` as well so that should be added back here as well.
```suggestion
				Optional:  true,
				Computed: true,
```